templates/kubernetes/terraform/modules/kubernetes/vpn.tf

```diff
@@ -0,0 +1,212 @@
+  name = local.server_privatekey_name
+}
+data "aws_secretsmanager_secret_version" "vpn_private_key" {
+  secret_id = data.aws_secretsmanager_secret.vpn_private_key.id
+}
+
+resource "kubernetes_namespace" "vpn_namespace" {
+  metadata {
+    name = local.namespace
+  }
+}
+
+resource "kubernetes_secret" "vpn_private_key" {
+  metadata {
+    name = "wg-secret"
+    namespace = local.namespace
+  }
+
+  data = {
+    privatekey = jsondecode(data.aws_secretsmanager_secret_version.vpn_private_key.secret_string)["key"]
+  }
+
+  type = "Opaque"
+}
+
+resource "kubernetes_config_map" "vpn_configmap" {
+  metadata {
+    name = "wg-configmap"
+    namespace = local.namespace
+  }
+
+  data = {
+    "wg0.conf" = "[Interface]\nAddress = ${local.server_address}\nListenPort = 51820\nPostUp = wg set wg0 private-key /etc/wireguard/privatekey && iptables -A FORWARD -s ${local.server_address} -d ${local.destination_subnets} -j ACCEPT && iptables -A FORWARD -s ${local.server_address} -j DROP && iptables -t nat -A POSTROUTING -s ${local.server_address} -o eth0 -j MASQUERADE\nPostDown = iptables -D FORWARD -s ${local.server_address} -d ${local.destination_subnets} -j ACCEPT && iptables -D FORWARD -s ${local.server_address} -j DROP && iptables -t nat -D POSTROUTING -s ${local.destination_subnets} -o eth0 -j MASQUERADE\n\n${local.client_peers}\n"
+  }
+}
+
+resource "kubernetes_service" "wireguard" {
+  metadata {
+    name = "wireguard"
+    namespace = local.namespace
+
+    labels = {
+      app = "wireguard"
+    }
+
+    annotations = {
+      "service.beta.kubernetes.io/aws-load-balancer-type" = "nlb"
+    }
+  }
+
+  spec {
+    port {
+      name        = "wg"
+      protocol    = "UDP"
+      port        = 51820
+      target_port = "51820"
+    }
+
+    selector = {
+      app = "wireguard"
+    }
+
+    type = "LoadBalancer"
+    external_traffic_policy = "Local"
+  }
+}
+
+resource "kubernetes_deployment" "wireguard" {
+  metadata {
+    name = "wireguard"
+    namespace = local.namespace
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = "wireguard"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "wireguard"
+        }
+      }
+
+      spec {
+        volume {
+          name = "cfgmap"
+
+          config_map {
+            name = "wg-configmap"
+          }
+        }
+
+        volume {
+          name = "secret"
+
+          secret {
+            secret_name = "wg-secret"
+          }
+        }
+
+        init_container {
+          name    = "sysctls"
+          image   = "busybox"
+          command = ["sh", "-c", "sysctl -w net.ipv4.ip_forward=1 && sysctl -w net.ipv4.conf.all.forwarding=1"]
+
+          security_context {
+            capabilities {
+              add = ["NET_ADMIN"]
+            }
+
+            privileged = true
+          }
+        }
+
+        container {
+          name    = "wireguard"
+          image   = "masipcat/wireguard-go:latest"
```

Comment:
Let's use a specific version, rather than `latest`. Also, we will probably want to mirror this image rather than using theirs, so we have control over the contents.

Reply:
Bill, the previous version was 5 months ago. Better we do a mirror from the latest.

Reply:
done with newly forked "commitdev/wireguard-go:0.0.1" and confirmed working